### PR TITLE
REST API tests for cloud_networks queries

### PR DESCRIPTION
**Pull request by mkoura**
__Adding tests__ for cloud_networks queries.

{{pytest: '''cfme/tests/cloud/test_providers.py::TestProvidersRESTAPI'' -v'}}

### CODE REVIEW

**Comment by mkoura** [comment]
@mfalesni

**Comment by dajoRH** [comment]
Lint report for commit a84646e83da1f26e9aeb8cf482895efb7981fabd:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

File: cfme/tests/cloud/test_providers.py [hunk 1]
```diff
@@ -311,3 +311,39 @@ def test_openstack_provider_has_api_version():
...
+
+        Metadata:
+            test_flag: rest
+        """
+        if from_detail:
+            networks = rest_api.collections.providers.get(name=setup_a_provider.name).cloud_networks
+        else:
+            networks = rest_api.collections.cloud_networks
+        assert len(networks) > 0
+        assert len(networks) == networks.subcount
```
Comment (psav) [inline cfme/tests/cloud/test_providers.py]: So, what is network.subcount == 0, perhaps we should just assert that it == the subcount?

Comment (mkoura) [inline cfme/tests/cloud/test_providers.py]: @psav There needs to be at least one cloud network present for this test to be valid (checking for "enabled":true, and "CloudNetwork" type). The first assert is there to make it obvious what the problem is if that's not the case.

**Comment by psav** [comment]
Some failures in PRT

**Comment by dajoRH** [comment]
Lint report for commit a0bfc0bc700f9a21aae6546fadea995e59f0be2a:

Lint OK! :cake: :punch: :cookie: .


**No commit flaws detected**.
<mergeable/>

*CFME QE Bot*

**Pull request merged**